mitosheet/src/mito/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -99,6 +160,76 @@ const GraphSidebar = (props: {
             }
             const executeScript = new Function(graphOutput.graphScript);
             executeScript()
+            const div: any = document.getElementById(graphOutput.graphHTML.split('id="')[1].split('"')[0])
+            if (div === null) {
+                return;
+            }
+            div.on('plotly_click', () => {
+                setSelectedGraphElement(null)
+            });
+            
+            // Main Title
+            const gtitle = div.getElementsByClassName('g-gtitle')[0]
+            const xtitle = div.getElementsByClassName('g-xtitle')[0]
+            const ytitle = div.getElementsByClassName('g-ytitle')[0]
+                            
+            // First, add the style to make it clickable with pointer-events: all
+            gtitle.style.pointerEvents = 'all'
+            xtitle.style.pointerEvents = 'all'
+            ytitle.style.pointerEvents = 'all'
+
+            /**
+             * Set selected graph element when clicked
+             */
+            gtitle.addEventListener('click', () => {
+                setSelectedGraphElement({
+                    element: 'gtitle',
+                    displayPopup: false
+                })
+            })
+            xtitle.addEventListener('click', () => {
+                setSelectedGraphElement({
+                    element: 'xtitle',
+                    displayPopup: false
+                });
+            })
+            ytitle.addEventListener('click', () => {
+                setSelectedGraphElement({
+                    element: 'ytitle',
+                    displayPopup: false
+                })
+            })
+
+            /**
+             * Open popup when double clicked
+             */
+            gtitle.addEventListener('dblclick', () => {
+                setSelectedGraphElement({
+                    element: 'gtitle',
+                    xPosition: gtitle.getBoundingClientRect().left,
+                    yPosition: gtitle.getBoundingClientRect().top + 30,
+                    displayPopup: true
```

Comment:
Feels like there's duplicated state here. Perhaps the better state is `editingPopupPosition : {xPosition, yPosition} | undefined` -- above, you set it to undefined, and below you have x and y.